### PR TITLE
17921 Update TransactionAlerts with BannerAlert

### LINKED ISSUE #17921: Update TransactionAlerts with BannerAlert
### Description
Replace `ActionableMessage` with `BannerAlert` in `ui/components/app/transaction-alerts/transaction-alerts.js`

## Before
Screenshot 2023-02-27 at 6 04 02 PM

## After

TBC

### Technical Description
- Update `ActionableMessage` with `BannerAlert` https://metamask.github.io/metamask-storybook/?path=/docs/components-componentlibrary-banneralert--default-story
- Update `Typography` with `Text`(If needed it probably wont be) https://metamask.github.io/metamask-storybook/?path=/docs/components-componentlibrary-text--default-story

### Related
- https://github.com/MetaMask/metamask-extension/issues/17873
- https://github.com/MetaMask/metamask-extension/pull/17738

### PR DESCRIPTION
**Pull request by flexa-rob**
## Explanation

Replace ActionableMessage with BannerAlert in ui/components/app/transaction-alerts/transaction-alerts.js

Fixes: #17921

### Before

image

### After

image

## Manual Testing Steps

Begin a transaction while network is congested, or network congestion is simulated.

## Pre-merge author checklist

- [ ] I've clearly explained:
  - [ ] What problem this PR is solving
  - [ ] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/components/app/transaction-alerts/transaction-alerts.js [hunk 1]
```diff
@@ -30,74 +34,61 @@ const TransactionAlerts = ({
...
+          severity={SEVERITIES.WARNING}
+        >
+          <Typography
+            align="left"
+            margin={0}
+            tag={TypographyVariant.paragraph}
+            variant={TypographyVariant.H7}
+          >
+            {t('lowPriorityMessage')}
+          </Typography>
```
Comment (georgewrmarshall) [inline ui/components/app/transaction-alerts/transaction-alerts.js:78]: I think we can remove the Typography component all together?
```suggestion
          {t('lowPriorityMessage')}
```

File: ui/components/app/transaction-alerts/transaction-alerts.js [hunk 2]
```diff
@@ -30,74 +34,61 @@ const TransactionAlerts = ({
...
-        />
+        <BannerAlert severity={SEVERITIES.WARNING}>
+          <Typography
+            align="left"
+            margin={0}
+            tag={TypographyVariant.paragraph}
+            variant={TypographyVariant.H7}
+          >
+            {t('networkIsBusy')}
+          </Typography>
```
Comment (georgewrmarshall) [inline ui/components/app/transaction-alerts/transaction-alerts.js:90]: Same here
```suggestion
          {t('networkIsBusy')}
```

**Review by georgewrmarshall — CHANGES_REQUESTED** [review]
Code is looking pretty good! Left some small suggestions would be great to get some stories for these components for easier visual review and iterations in future

File: ui/components/app/transaction-alerts/transaction-alerts.js (hunk 1 above)
Comment (flexa-rob) [reply ui/components/app/transaction-alerts/transaction-alerts.js:78]: Cool, I'll remove those blocks and see how it looks. I was originally thinking of adding stories too, just got complicated with state - I'll look into it again today and try figure it out.

File: ui/components/app/transaction-alerts/transaction-alerts.js [hunk 3]
```diff
@@ -30,74 +34,61 @@ const TransactionAlerts = ({
...
-          type="warning"
-        />
+        <BannerAlert severity={SEVERITIES.WARNING}>
+          <Typography
+            align="left"
+            className="transaction-alerts__pending-transactions"
+            margin={0}
+            tag={TypographyVariant.paragraph}
+            variant={TypographyVariant.H7}
+          >
```
Comment (flexa-rob) [inline ui/components/app/transaction-alerts/transaction-alerts.js:44]: @georgewrmarshall - should I remove this `Typography` also (middle alert in the following screenshot)?

image

It has embedded links etc, so will need some styling fixes:

image

Comment (georgewrmarshall) [reply ui/components/app/transaction-alerts/transaction-alerts.js:43]: Yes! The `Typography` has been deprecated. You could do something like this. (Note use of ButtonLink here)

`import { BannerAlert, Text, ButtonLink } from '../../component-library';`

```
<BannerAlert severity={SEVERITIES.WARNING}>
          <Text as="p">
            <strong>
              {pendingTransactions?.length === 1
                ? t('pendingTransactionSingle', [pendingTransactions?.length])
                : t('pendingTransactionMultiple', [
                    pendingTransactions?.length,
                  ])}
            </strong>{' '}
            {t('pendingTransactionInfo')}
            {t('learnCancelSpeeedup', [
              <ButtonLink
                key="cancelSpeedUpInfo"
                href={ZENDESK_URLS.SPEEDUP_CANCEL}
                rel="noopener noreferrer"
                target="_blank"
              >
                {t('cancelSpeedUp')}
              </>,
            ])}
          </Text>
        </BannerAlert>
```

File: ui/components/app/transaction-alerts/transaction-alerts.js (hunk 1 above)
Comment (georgewrmarshall) [reply ui/components/app/transaction-alerts/transaction-alerts.js]: Do a search for `configureStore` and only accept `*.stories.js` it should give you some examples

Comment (flexa-rob) [reply ui/components/app/transaction-alerts/transaction-alerts.js]: ok, thanks -  I was able to get 3 of the four alerts to show by manipulating state. I'll work on getting the pending transactions alert state tomorrow.

image

File: ui/components/app/transaction-alerts/transaction-alerts.js (hunk 3 above)
Comment (flexa-rob) [reply ui/components/app/transaction-alerts/transaction-alerts.js]: done

File: ui/components/app/transaction-alerts/transaction-alerts.js (hunk 2 above)
Comment (flexa-rob) [reply ui/components/app/transaction-alerts/transaction-alerts.js]: done

File: ui/components/app/transaction-alerts/transaction-alerts.js (hunk 1 above)
Comment (flexa-rob) [reply ui/components/app/transaction-alerts/transaction-alerts.js]: Pending transactions alert is now in, with separated stories for each alert scenario

image

**Review by georgewrmarshall — APPROVED** [review]
Really really nice. I think we can use the `ui/components/app/transaction-alerts/transaction-alerts.stories.js` as a great example of mocking stories what need complex state 💯 🚀

**Comment by flexa-rob** [comment]
> Really really nice. I think we can use the `ui/components/app/transaction-alerts/transaction-alerts.stories.js` as a great example of mocking stories what need complex state 100 rocket

Thanks @georgewrmarshall 

I just need 1 more approval on these - is there someone else you can recommend?

**Review by VSaric — APPROVED** [review]
LGTM!

**Review by SaraCheikh — APPROVED** [review]

**Comment by flexa-rob** [comment]
@SaraCheikh , @georgewrmarshall , @VSaric - is this ok to merge now? Would you like to do it, or should I?

(or do you want to wait and merge this together with #17937 and/or #17786 ?)

**Comment by georgewrmarshall** [comment]
Fine to merge @flexa-rob

**Pull request merged**